src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,140 @@
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.getKey(), key)) {
+                    return currentNode.getValue();
+                }
+                currentNode = currentNode.nextItemInBucket;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndexByHashCode(Object key) {
+        if (key == null) {
+            return 0;
+        }
+        int index = (key.hashCode() % bucketsArray.length);
+        return index >= 0 ? index : index * -1;
+    }
+
+    private void resize() {
+        threshold = (int) (bucketsArray.length * LOADING_FACTOR);
+        Node<K, V>[] oldBucketsArray = bucketsArray;
+        bucketsArray = new Node[bucketsArray.length * RESIZE_MULTIPLIER];
+        for (Node<K, V> node : oldBucketsArray) {
+            if (node != null) {
+                Node<K, V> currentNode = node;
+                while (currentNode != null) {
+                    addElement(new Node<K, V>(currentNode.key, currentNode.value));
+                    currentNode = currentNode.nextItemInBucket;
+
+                }
+            }
+        }
+    }
+
+    private void addElement(Node<K, V> node) {
+        int index = node.getKey() == null ? 0 : getIndexByHashCode(node.getKey());
+        if (bucketsArray[index] == null) {
+            bucketsArray[index] = node;
+            return;
+        }
+        if (Objects.equals(bucketsArray[index].getKey(), node.getKey())) {
+            bucketsArray[index].setValue(node.getValue());
+        } else {
+            Node<K, V> currentNode = getLastOrEqualNode(bucketsArray[index], node.getKey());
+            node.nextItemInBucket = currentNode.nextItemInBucket;
+            currentNode.nextItemInBucket = node;
+        }
+    }
+
+    private Node<K, V> getLastOrEqualNode(Node<K,V> currentNode, K key) {
+        while (currentNode.nextItemInBucket != null) {
+            if (Objects.equals(currentNode.nextItemInBucket.getKey(), key)) {
+                return currentNode;
+            }
+            currentNode = currentNode.nextItemInBucket;
+        }
+        return currentNode;
+    }
+
+    private boolean keyIsNew(K key) {
+        if (bucketsArray[getIndexByHashCode(key)] == null) {
+            return true;
+        }
+        Node<K, V> current = bucketsArray[getIndexByHashCode(key)];
+        while (current != null) {
+            if (Objects.equals(key, current.key)) {
+                return false;
+            }
+            current = current.nextItemInBucket;
+        }
+        return true;
+
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> nextItemInBucket;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.nextItemInBucket = null;
```

Comment:
```suggestion
            this.key = key;
            this.value = value;
```
`nextItemInBucket` is null by default